### ISSUE TITLE
以tlx.nn.Sequential()为基类的网络类必须写forward()方法，不写有可能跳过某些层，paddle的不用写

### ISSUE THREAD

**Issue by moshizhiyin** [tensorlayer/TensorLayerX#33]
debug时发现可能是以Sequential()为基类的网络类无法获取网络层参数信息”Unable to get repr for <class '__main__.Conv2d_BN'>“
，虽然不影响正常的参数传递，但是经过
        self.layer_list = list(self._sub_layers.values())
之后就不能正常工作了

**Comment by hanjr92** [comment]
方便发一下代码么。

**Comment by moshizhiyin** [comment]
涉及的几个类
class Conv2d_BN(nn.Sequential):
    def __init__(self,
                 a,
                 b,
                 ks=1,
                 stride=1,
                 pad=0,
                 dilation=1,
                 groups=1,
                 bn_weight_init=1,
                 resolution=-10000):
        super().__init__()
        self.add_sublayer(
            'c',
            nn.GroupConv2d(
                in_channels=a,
                out_channels=b,
                kernel_size=ks,
                stride=stride,
                padding=pad,
                dilation=dilation,
                n_group=groups,
                data_format="channels_first",
                b_init=None))
        bn = nn.BatchNorm2d(num_features=b, data_format="channels_first")
        ones_(bn.gamma)
        zeros_(bn.beta)
        self.add_sublayer('bn', bn)


class Linear_BN(nn.Sequential):
    def __init__(self, a, b, bn_weight_init=1):
        super().__init__()
        self.add_sublayer('c', nn.Linear(in_features=a, out_features=b, b_init=None))
        bn = nn.BatchNorm1d(num_features=b, data_format="channels_first")
        if bn_weight_init == 0:
            zeros_(bn.gamma)
        else:
            ones_(bn.gamma)
        zeros_(bn.beta)
        self.add_sublayer('bn', bn)

    def forward(self, x):
        l, bn = self._sub_layers.values()
        x = l(x)
        return tlx.reshape(bn(x.flatten(0, 1)), x.shape)


class BN_Linear(nn.Sequential):
    def __init__(self, a, b, bias=True, std=0.02):
        super().__init__()
        self.add_sublayer('bn', nn.BatchNorm1d(num_features=a, data_format="channels_first"))
        if bias:
            l = nn.Linear(in_features=a, out_features=b)
        else:
            l = nn.Linear(in_features=a, out_features=b, b_init=None)
        trunc_normal_(l.weights)
        if bias:
            zeros_(l.biases)
        self.add_sublayer('l', l)



def b16(n, activation, resolution=224):
    return nn.Sequential(
        Conv2d_BN(
            3, n // 8, 3, 2, 1, resolution=resolution),
        activation(),
        Conv2d_BN(
            n // 8, n // 4, 3, 2, 1, resolution=resolution // 2),
        activation(),
        Conv2d_BN(
            n // 4, n // 2, 3, 2, 1, resolution=resolution // 4),
        activation(),
        Conv2d_BN(
            n // 2, n, 3, 2, 1, resolution=resolution // 8))

全部的有点长，方便给一下邮箱吗

**Comment by hanjr92** [comment]
方便的话，把.py文件发到 jiaronghan@outlook.com

**Comment by moshizhiyin** [comment]
> 方便的话，把.py文件发到 [jiaronghan@outlook.com](mailto:jiaronghan@outlook.com)

发了，后面补发了测试图片和需要的方法
